lib/googleauth/signet.rb

```diff
@@ -77,6 +77,22 @@ def notify_refresh_listeners
           block.call(self)
         end
       end
+
+      def retry_with_error(retry_count = 5)
+        begin
+          yield
+        rescue => e
+          raise e if e.is_a?(Signet::AuthorizationError) || e.is_a?(Signet::ParseError)
+
+          if retry_count > 0
+            retry_count -= 1
+            retry
```

Comment:
I've usually seen retries with an accompanying backoff. So there is an increasing delay between each retry. That seems to be a common Googleism.